pkg/skaffold/deploy/helm.go

```diff
@@ -182,13 +187,22 @@ func (h *HelmDeployer) Dependencies() ([]string, error) {
 
 // Cleanup deletes what was deployed by calling Deploy.
 func (h *HelmDeployer) Cleanup(ctx context.Context, out io.Writer) error {
+	hv, err := h.binVer(ctx)
+	if err != nil {
+		return errors.Wrap(err, "binary version")
+	}
+
 	for _, r := range h.Releases {
 		releaseName, err := expand(r.Name, nil)
 		if err != nil {
 			return errors.Wrap(err, "cannot parse the release name template")
 		}
 
-		if err := h.exec(ctx, out, false, "delete", releaseName, "--purge"); err != nil {
+		args := []string{"delete", releaseName}
+		if hv.LT(semver.MustParse("3.0.0-beta.0")) {
```

Comment:
since `semver.MustParse("3.0.0-beta.0")` is used several times, I would make it a constant

Reply:
Done.